docs/design/rubygems.md

```diff
@@ -0,0 +1,522 @@
+build to fail.
+
+##### Multiple Gems in a single repository
+
+A single repository can hold multiple Gems, and those can be imported as dependencies. When this happens, Bundler still
+expects a single clone to be made. Here's an example of how multiple gems imported from a single repository+revision
+looks like in the `Gemfile.lock`:
+
+```
+GIT
+  remote: https://github.com/chatwoot/azure-storage-ruby
+  revision: 9957cf899d33a285b5dfe15bdb875292398e392b
+  branch: chatwoot
+  specs:
+    azure-storage-blob (2.0.3)
+      azure-storage-common (~> 2.0)
+      nokogiri (~> 1, >= 1.10.8)
+    azure-storage-common (2.0.4)
+      faraday (~> 2.0)
+      faraday-follow_redirects (~> 0.3.0)
+      faraday-net_http_persistent (~> 2.0)
+      net-http-persistent (~> 4.0)
+      nokogiri (~> 1, >= 1.10.8)
+```
+
+### Out of scope
+
+#### Plugins
+Bundler has support for using [plugins](https://bundler.io/guides/bundler_plugins.html), which allows users to extend
+Bundler's functionality in any way that they seem fit. Since this can open the possibility for security issues, plugins
+will not be supported by Cachi2.
+
+Since we're not proposing the direct usage of Bundler to fetch the dependencies, no other actions are needed in the
+prefetch phase, existing plugin definitions will be ignored.
+
+#### Pre-compiled binaries
+For the initial implementation, we're aiming to provide support only for plain Ruby gems (which are idenfied as `RUBY`
+in the `PLATFORMS` section of the `Gemfile.lock`). This means that we need to explicitly forbid the processing of any
+repository that has references to any platform besides `RUBY`.
+
+To achieve this, we can implement a check in Cachi2 that will cause the request to fail if any other platform is
+specified in the `Gemfile.lock`. This way, we can avoid downloading any pre-compiled content.
+
+We should probably implement support for pre-compiled binaries as a follow-up feature, similarly to what was done with
+[pip wheels](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#distribution-formats).
+
+#### Checksum verification
+Since checksums in the `Gemlock.file` is still a feature in development (see [checksums](#dependency-checksums)), we
+can postpone implementing support for it until the feature is delivered.
+
+We need to decide if we will report all dependencies as having missing checksums in the SBOM, or not.
+
+#### Alternate registries
+Users can specify different registries by leveraging the `source` keyword on the `Gemfile`. This setting can be global
+or target to a specific group of Gems. See the [official docs](https://bundler.io/guides/gemfile.html#gemfiles) for
+more info.
+
+We can likely just support the use case of the global `source` being set to `'https://rubygems.org'` (the default)
+value. Cachi2 will need to check for ny changes on the global `source` or additional `source` blocks added to the
+`Gemfile`, and fail the request in case they're found.
```

Comment:
It's also reflected in Gemfile.lock, right? Technically there's no reason not to support it

```Gemfile.lock
GEM
 remote: ...
```

Reply:
You're right, I think I went on the lazy side here. We can simply do:

```
url = f"{gem.source}/gems/{gem.name}-{gem.version}.gem"
```

If we assume that the alternate registry will behave exactly like "rubygems.org".